sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala

```diff
@@ -343,6 +343,8 @@ class SQLListenerMemoryLeakSuite extends SparkFunSuite {
       .set("spark.sql.ui.retainedExecutions", "50") // Set it to 50 to run this test quickly
     val sc = new SparkContext(conf)
     try {
+      // Clear the sql listener created by a previous test suite.
+      SQLContext.clearSqlListener()
```

Comment:
[Over on the original PR, I commented](https://github.com/apache/spark/pull/9297/files#r46203496) to ask why `SQLContext.sqlListener` needs to be an `AtomicReference[SQLListener]` instead of an `AtomicBoolean` or some other sort of atomic primitive. As far as I can tell, we never access any methods or fields of the `sqlListener` that's stored here, so if we only need to set something for compare-and-swap purposes then I think we shouldn't use an `AtomicReference`, thereby avoiding the GC issues that it causes.